repos/spack_repo/builtin/packages/scotch/package.py

```diff
@@ -63,6 +63,13 @@ class Scotch(CMakePackage, MakefilePackage):
         when="@7.0.1",
         description="Link error handling library to libscotch/libptscotch",
     )
+    variant(
```

Comment:
Is this variant only valid for `@7.0.7:` as suggested below? If so, then shouldn't the `variant` should have a `when=` clause?

I also think that version comparisons can be done in a much simpler way, namely `if self.spec.satisfies("@7.0.4:")` etc.